contracts/components/staking/FortaStaking.sol

```diff
@@ -397,58 +400,98 @@ contract FortaStaking is BaseComponentUpgradeable, ERC1155SupplyUpgradeable, Sub
+        uint256 slashFromActive = _slash(activeSharesId, subjectType, subject, stakeValue);
+        uint256 slashedFromDelegators;
+        if (getSubjectTypeAgency(subjectType) == SubjectStakeAgency.DELEGATED) {
+            _allocator.withdrawAllocation(activeSharesId, subjectType, subject, slashFromActive);
+
+            if (slashDelegatorsPercent > 0) {
+                slashedFromDelegators = _slashDelegator(getDelegatorSubjectType(subjectType), subject, stakeValue);
+            }
+        }
+
+        if (proposerPercent > 0) {
+            if (proposer == address(0)) revert ZeroAddress("proposer");
+            uint256 proposerShare = Math.mulDiv(stakeValue + slashedFromDelegators, proposerPercent, HUNDRED_PERCENT);
+            SafeERC20.safeTransfer(stakedToken, proposer, proposerShare);
+            emit SlashedShareSent(subjectType, subject, proposer, proposerShare);
+            SafeERC20.safeTransfer(stakedToken, _treasury, stakeValue + slashedFromDelegators - proposerShare);
+        } else {
+            SafeERC20.safeTransfer(stakedToken, _treasury, stakeValue + slashedFromDelegators);
+        }
+
+        return stakeValue;
+    }
+
+    /**
+     * @notice burns slashed stake from active and/or inactive stake for subjectType/subject.
+     * @param activeSharesId ERC1155 id of the shares being slashed
+     * @param subjectType type id of Stake Subject. See SubjectTypeValidator.sol
+     * @param subject id identifying subject (external to FortaStaking).
+     * @param stakeValue amount of staked token to be slashed.
+     */
+    function _slash(
+        uint256 activeSharesId,
+        uint8 subjectType,
+        uint256 subject,
+        uint256 stakeValue
+    ) private returns (uint256 slashFromActive) {
         uint256 activeStake = _activeStake.balanceOf(activeSharesId);
         uint256 inactiveStake = _inactiveStake.balanceOf(FortaStakingUtils.activeToInactive(activeSharesId));
-
         // We set the slash limit at 90% of the stake, so new depositors on slashed pools (with now 0 stake) won't mint
         // an amounts of shares so big that they might cause overflows.
         // New shares = pool shares * new staked amount / pool stake
         // See deposit and stakeToActiveShares methods.
-        uint256 maxSlashableStake = Math.mulDiv(activeStake + inactiveStake, _stakingParameters.maxSlashableStakePercent(), HUNDRED_PERCENT);
+        uint256 maxSlashableStake = Math.mulDiv(activeStake + inactiveStake, MAX_SLASHABLE_PERCENT, HUNDRED_PERCENT);
+
         if (stakeValue > maxSlashableStake) revert SlashingOver90Percent();
 
-        uint256 slashFromActive = Math.mulDiv(activeStake, stakeValue, activeStake + inactiveStake);
+        slashFromActive = Math.mulDiv(activeStake, stakeValue, activeStake + inactiveStake);
         uint256 slashFromInactive = stakeValue - slashFromActive;
-        stakeValue = slashFromActive + slashFromInactive;
```

Comment:
Why was this line deleted or not moved to the `slash` function?

Reply:
cause it is not really necessary. I think it's a left over on a comment I have, `stakeValue` is defined earlier, and it _is_ `slashFromActive + slashFromInactive`, since those are derived from stakeValue.